File: design-patterns/java/mediator/FormMediator.java
public class FormMediator implements UIMediator {
    private TextField usernameField;
    private TextField passwordField;
    private Button loginButton;
    private Label statusLabel;

    public void setUsernameField(TextField usernameField) {
        this.usernameField = usernameField;
    }

    public void setPasswordField(TextField passwordField) {
        this.passwordField = passwordField;
    }

    public void setLoginButton(Button loginButton) {
        this.loginButton = loginButton;
    }

    public void setStatusLabel(Label statusLabel) {
        this.statusLabel = statusLabel;
    }

    @Override
    public void componentChanged(UIComponent component) {
        if (component == usernameField || component == passwordField) {
            boolean enableButton = !usernameField.getText().isEmpty() && !passwordField.getText().isEmpty();
            loginButton.setEnabled(enableButton);
        } else if (component == loginButton) {
            String username = usernameField.getText();
            String password = passwordField.getText();

            if ("admin".equals(username) && "1234".equals(password)) {
                statusLabel.setText("✅ Login successful!");
            } else {
                statusLabel.setText("❌ Invalid credentials.");
            }
        }
    }
}

File: design-patterns/cpp/mediator/form_mediator.cpp
#include "form_mediator.h"
#include <iostream>

void FormMediator::setUsernameField(TextField* field) {
    usernameField = field;
}

void FormMediator::setPasswordField(TextField* field) {
    passwordField = field;
}

void FormMediator::setLoginButton(Button* button) {
    loginButton = button;
}

void FormMediator::setStatusLabel(Label* label) {
    statusLabel = label;
}

void FormMediator::componentChanged(UIComponent* component) {
    if (component == usernameField || component == passwordField) {
        bool enableButton = !usernameField->getText().empty() && !passwordField->getText().empty();
        loginButton->setEnabled(enableButton);
    } else if (component == loginButton) {
        const std::string& username = usernameField->getText();
        const std::string& password = passwordField->getText();

        if (username == "admin" && password == "1234") {
            statusLabel->setText("✅ Login successful!");
        } else {
            statusLabel->setText("❌ Invalid credentials.");
        }
    }
} 